src/panel-ladder.tsx

```diff
@@ -0,0 +1,182 @@
+		for (const [key, format] of Object.entries(BattleFormats)) {
+			if (!format.rated || !format.searchShow) continue;
+			if (format.section !== currentSection) {
+				if (formats.length > 0) {
+					sections.push(<preact.Fragment key={currentSection}><h3>{currentSection}</h3><ul style="list-style:none;margin:0;padding:0">{formats}</ul></preact.Fragment>);
+					formats = [];
+				}
+				currentSection = format.section;
+			}
+			formats.push(<li key={key} style="margin:5px"><button name="selectFormat" value={key} class="button" style="width:320px;height:30px;text-align:left;font:12pt Verdana" onClick={this.handleSetFormat(key)}>{BattleLog.escapeFormat(format.id)}</button></li>);
+		}
+		return <>
+			{sections}
+		</>;
+	};
+	ShowFormatList = () => {
+		return <>
+			<p>See a user's ranking with <a class="button" href={`/${Config.routes.users}/`} target="_blank">User lookup</a></p>
+			<this.Notice/>
+			<p>(btw if you couldn't tell the ladder screens aren't done yet; they'll look nicer than this once I'm done.)</p>
+			<p><button name="selectFormat" value="help" class="button" onClick={this.setShowHelp(true)}><i class="fa fa-info-circle"></i> How the ladder works</button></p>
+			<this.BattleFormatList/>
+		</>;
+	};
+	FormatListButton = () => {
+		return <button name="selectFormat" onClick={this.handleSetFormat(undefined)}><i class="fa fa-chevron-left"></i> Format List</button>;
+	};
+	ShowFormat = () => {
+		const { teams } = PS;
+		const { room } = this.props;
+		const { searchValue, lastSearch } = this.state;
+		const selectedFormat = room.selectedFormat as string;
+		if (room.ladderData === undefined) {
+			return <div class="ladder pad"><p><this.FormatListButton/></p><p><em>Loading...</em></p></div>;
+		} else if (teams.usesLocalLadder) {
+			return <div class="ladder pad">
+				<p><this.FormatListButton/></p>
+				<div dangerouslySetInnerHTML={{__html: room.ladderData}}></div>
+			</div>;
+		}
+		return <div class="ladder pad">
+			<p><this.FormatListButton/></p><p><button class="button" name="refresh" onClick={this.handleRequestLadderData}><i class="fa fa-refresh"></i> Refresh</button>
+			<form class="search" onSubmit={this.submitSearch}><input type="text" name="searchValue" class="textbox searchinput" value={BattleLog.escapeHTML(searchValue)} placeholder="username prefix" onChange={this.setSearchValue} /><button type="submit"> Search</button></form></p>
+			<h3>{BattleLog.escapeFormat(selectedFormat)} Top {BattleLog.escapeHTML(lastSearch ? `- '${lastSearch}'` : '500')}</h3>
+			<div dangerouslySetInnerHTML={{__html: room.ladderData}}></div>
+		</div>; // That's right, danger!
+	};
+	render() {
+		const room = this.props.room;
+		const { showHelp } = this.state;
+		return <PSPanelWrapper room={room} scrollable>
+			<div class="ladder pad">
+				{showHelp && <this.Help/>}
+				{!showHelp && room.selectedFormat === undefined && <this.ShowFormatList/>}
+				{!showHelp && room.selectedFormat !== undefined && <this.ShowFormat/>}
```

Comment:
I wonder if this should be separate rooms with separate room IDs.

Like:

`ladder`, `ladder-gen8ou`, `ladderhelp`.

That's how the new teambuilder is set up and I think it'd be better.

Reply:
Complete